dcicutils/glacier_utils.py

```diff
@@ -0,0 +1,321 @@
+        """
+        try:  # extract temporary location by heading object
+            response = self.s3.head_object(Bucket=bucket, Key=key)
+            restore = response.get('Restore')
+            if restore is None:
+                PRINT(f'Object {bucket}/{key} is not currently being restored from Glacier')
+                return None
+            if 'ongoing-request="false"' not in restore:
+                PRINT(f'Object {bucket}/{key} is still being restored from Glacier')
+                return None
+            restore_path = response.get('RestoreOutputPath')
+            if restore_path is None:
+                PRINT(f'Error: Could not determine the temporary location of restored object {bucket}/{key}')
+                return None
+            return restore
+        except Exception as e:
+            PRINT(f'Error copying object {bucket}/{key} back to its original location in S3: {str(e)}')
+            return None
+
+    def patch_file_lifecycle_status_to_standard(self, atid: str, status: str,
+                                                s3_lifecycle_status: str = 'standard') -> dict:
+        """ Patches the File @id object to update 3 things
+                1. status denoted by the status argument (usually uploaded)
+                2. s3_lifecycle_status to 'standard' by default
+                3. s3_lifecycle_category deleted (so lifecycle transition doesn't occur again)
+
+        :param atid: object resource path to update
+        :param status: status to replace at top level
+        :param s3_lifecycle_status: life cycle status to replace on file, typically standard as that is the default
+        :return: response from patch_metadata
+        """
+        return patch_metadata(
+            {
+                'status': status,
+                's3_lifecycle_status': s3_lifecycle_status
+            },
+            atid,
+            key=self.env_key, ff_env=self.env_name,
+            add_on='?delete_fields=s3_lifecycle_category'
+        )
+
+    def delete_glaciered_object_versions(self, bucket: str, key: str, delete_all_versions: bool = False) -> bool:
+        """ Deletes glaciered object versions of the given bucket/key, clearing all versions in glacier if the
+            delete_all_versions flag is specified
+
+        :param bucket: bucket location containing key
+        :param key: file name in s3 to delete
+        :param delete_all_versions: whether or not to delete all glacier versions or just the most recent one
+        :return: True if success or False if failed
+        """
+        try:
+            # Retrieve the object versions for the key
+            response = self.s3.list_object_versions(Bucket=bucket, Prefix=key)
+            versions = response.get('Versions', [])
+            versions.sort(key=lambda x: x['VersionId'], reverse=True)  # most recent version will be first
+
+            # Delete all glaciered versions if the flag is set
+            if delete_all_versions:
```

Comment:
Might be missing something here, isn't the same thing happening here regardless of the value of `delete_all_version`? For all versions, the storage class is checked, then the object is deleted and lines are logged.

Reply:
I've cleaned up this method considerably to address this comment and the above, thanks!